src/components/YourStore/index.tsx

```diff
@@ -13,58 +12,14 @@ export interface YourStoreProps {
   categories: GetCategoryReturn[];
 }
 
-export default function YourStore({ categories }: YourStoreProps) {
-  const [categoriesWithProducts, setCategoriesWithProducts] =
-    useState(categories);
-  const onlyCategoriesWithProducts = categoriesWithProducts.filter(
-    (category) => category.categoryProducts.length > 0
+export default function YourStore() {
+  const reduxCategories = useAppSelector(
```

Comment:
```suggestion
  const categories = useAppSelector(
```
I think just categories is a good name